### ISSUE TITLE
An error occurred while analyzing '...../.csproj' or '../.js' (Sonatype OSS Index Analyzer). 

### ISSUE THREAD

**Issue by milyas-salik** [dependency-check/DependencyCheck#3950]
Hi,

**Suddenly I started getting below exception in my Azure CI Task which uses OWASP Dependency check plugin to scan** 

I am using the latest version of OWASP dependency check.  Using Task to integrate OWASP dependency check from https://marketplace.visualstudio.com/items?itemName=InfoSupport.infosupport-owasp-dependecy-checker 

Logs:
```
[INFO] Finished RetireJS Analyzer (1 seconds)
[WARN] An error occurred while analyzing 'D:\Agents\Prd\AgentNameXYZ\_work\33\s\SRC\ProjectABC\gulp-config.js' (Sonatype OSS Index Analyzer).
[INFO] Finished Sonatype OSS Index Analyzer (10 seconds)
[INFO] Finished Vulnerability Suppression Analyzer (0 seconds)
[INFO] Finished Dependency Bundling Analyzer (0 seconds)
[INFO] Analysis Complete (15 seconds)
[INFO] Writing report to: D:\Agents\Prd\AgentNameXYZ\_work\_temp\dependency-scan-results\dependency-check-report.json
[ERROR] Could not execute .NET AssemblyAnalyzer
[ERROR] Failed to request component-reports
##[error]The process 'D:\Agents\Prd\AgentNameXYZe\_work\_tool\owasp-dependency-check\6.5.2\x64\dependency-check\bin\dependency-check.bat' failed with exit code 4294967282
##[section]Finishing: Execute OWASP Dependency Check

```

Log from another Project Pipeline

```

[INFO] Finished RetireJS Analyzer (0 seconds)
[WARN] An error occurred while analyzing 'D:\Agents\Tst\AgentNameABC\_work\1\s\SRC\DTS Solution\AbcServices\ProjectXYZ.csproj' (Sonatype OSS Index Analyzer).
[INFO] Finished Sonatype OSS Index Analyzer (10 seconds)
[INFO] Finished Vulnerability Suppression Analyzer (0 seconds)
[INFO] Finished Dependency Bundling Analyzer (0 seconds)
[INFO] Analysis Complete (13 seconds)
[INFO] Writing report to: D:\Agents\Tst\AgentNameABC\_work\_temp\dependency-scan-results\dependency-check-report.xml
[INFO] Writing report to: D:\Agents\Tst\AgentNameABC\_work\_temp\dependency-scan-results\dependency-check-report.html
[INFO] Writing report to: D:\Agents\Tst\AgentNameABC\_work\_temp\dependency-scan-results\dependency-check-report.json
[INFO] Writing report to: D:\Agents\Tst\AgentNameABC\_work\_temp\dependency-scan-results\dependency-check-report.csv
[INFO] Writing report to: D:\Agents\Tst\AgentNameABC\_work\_temp\dependency-scan-results\dependency-check-report.sarif
[INFO] Writing report to: D:\Agents\Tst\AgentNameABC\_work\_temp\dependency-scan-results\dependency-check-junit.xml
[ERROR] Failed to request component-reports
```

**Please do the needful**

**Comment by jeremylong** [comment]
OSS Index may have had an outage? i just ran a test and the services hosted by Sonatype that backs the OSS Index Analyzer is up and running now.  Are you still seeing the issue?

**Comment by milyas-salik** [comment]
@jeremylong  same issue not get fixed. 
Another question, I checked in the server dotnet --version its giving 2.2.110. 
Do you think upgrading to the latest .NET framework will fix it?

**Comment by jeremylong** [comment]
The OSS Index Analyzer error and the .NET error are unrelated - specifically:

```
[ERROR] Could not execute .NET AssemblyAnalyzer
[ERROR] Failed to request component-reports
```

As it looks like you may have dotnet code - currently ODC needs the dotnet core 3.1 runtime. Even if your project uses a different runtime it can still be analyzed.

To debug the Sonatype error we would need the verbose log (add `--log odc.log` to the command). My initial bet is that you are behind a proxy that is preventing you from connecting out to the OSS Index. In which case, either work with the networking team to allow the connection or disable the OSS Index Analyzer. The OSS Index Analyzer does report some vulnerabilities that are not contained in the NVD - but if you system is not allowed to connect out you may not have a choice and may need to disable the analyzer.

**Comment by milyas-salik** [comment]
@jeremylong : 

Yes, my project is .NET based and we are using .NET FW 4.7.2. What I understood is, it will still work in my case even my server (where the build agent is running for my Azure CI pipeline) does not have Dotnet core 3.1 run time.

If you can give me the specific URLs which plugin uses for OSS Index, I will take the help of a networking team to check and resolve the accessibility,.

I am not using the CMD console to scan with dependency-check.bat file. I am using the Task in Azure CI Build pipeline to automate it on each check in.
https://marketplace.visualstudio.com/items?itemName=InfoSupport.infosupport-owasp-dependecy-checker&ssr=false#qna

Can you tell me in my case, where to add `--log odc.log`  to get the verbose log.

<img width="848" alt="image" src="https://user-images.githubusercontent.com/79843909/148227134-6390b2aa-c55a-4311-b8ac-028deffafb76.png">

**Comment by aikebah** [comment]
the `--log odc.log` would go into the 'Additional arguments' section. For any support if your own attempts fail you should use the Q&A on the marketplace for that Azure pipeline plugin Task as the plugin maintainers will know best how to properly add the additional arguments in their configuration interface so that they are properly propagated to the CLI command that it triggers.

**Comment by jeremylong** [comment]
@milyas-salik in order to run the dotnet assembly analyzer - dotnet core 3.1 would need to be installed.

**Comment by milyas-salik** [comment]
@jeremylong please find the generated odc.log
[odc.log](https://github.com/jeremylong/DependencyCheck/files/7820480/odc.log)
.

**Comment by jeremylong** [comment]
Yeah - kind of what I expected:

```
org.owasp.dependencycheck.analyzer.exception.AnalysisException: Failed to request component-reports
Caused by: java.net.SocketTimeoutException: Connect timed out
```

To use the OSS Index Analyzer the tool connects out to https://ossindex.sonatype.org - basically, the same package information is sent that would be sent to Maven Central in a maven build and their service will return if there are known vulnerabilities in the dependency.

**Comment by milyas-salik** [comment]
@jeremylong  thank you so much. I am working on it with IT to resolve and also to install Dotnet core 3.1 run time for .NET assembly analyzer issue.

Another question, when we navigate to the Microsoft link for Dotnet core 3.1 there are basically 3 exes (asp.net, desktop, and .net), do we have to install them all ? I believe ASP.NET Core Runtime 3.1.22 is enough. Please advise.

https://dotnet.microsoft.com/en-us/download/dotnet/3.1

1. ASP.NET Core Runtime 3.1.22
2. .NET Desktop Runtime 3.1.22
3. .NET Runtime 3.1.22

**Comment by jeremylong** [comment]
You only need the .Net runtime.

**Comment by milyas-salik** [comment]
@jeremylong  I was able to resolve the OSS index issue, but not sure I have started with another issue. Attaching odc.log for reference.
```

2022-01-18 11:50:24,491 org.owasp.dependencycheck.App:210
ERROR - Failed to request component-reports
```
[odc.log](https://github.com/jeremylong/DependencyCheck/files/7887248/odc.log)

_Please do the needful._

**Comment by jeremylong** [comment]
Your network is likely also blocking `https://registry.npmjs.org/-/npm/v1/security/audits` which is required for the npm, yarn, and pnpm analyzers to work.

**Comment by milyas-salik** [comment]
Dear @jeremylong ,

Our IT has tried allowed the above URL but looks the issue is same.
```
DEBUG - unexpected error
org.owasp.dependencycheck.analyzer.exception.AnalysisException: Failed to request component-reports
	at org.owasp.dependencycheck.analyzer.OssIndexAnalyzer.analyzeDependency(OssIndexAnalyzer.java:151)
	at org.owasp.dependencycheck.analyzer.AbstractAnalyzer.analyze(AbstractAnalyzer.java:131)
	at org.owasp.dependencycheck.AnalysisTask.call(AnalysisTask.java:88)
	at org.owasp.dependencycheck.AnalysisTask.call(AnalysisTask.java:37)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1130)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:630)
	at java.base/java.lang.Thread.run(Thread.java:832)
Caused by: java.net.SocketTimeoutException: Connect timed out
	at java.base/sun.nio.ch.NioSocketImpl.timedFinishConnect(NioSocketImpl.java:546)
	at java.base/sun.nio.ch.NioSocketImpl.connect(NioSocketImpl.java:597)
	at java.base/java.net.Socket.connect(Socket.java:648)
	at java.base/sun.net.NetworkClient.doConnect(NetworkClient.java:177)
	at java.base/sun.net.www.http.HttpClient.openServer(HttpClient.java:474)
	at java.base/sun.net.www.http.HttpClient.openServer(HttpClient.java:569)
	at java.base/sun.net.www.protocol.https.HttpsClient.<init>(HttpsClient.java:265)
	at java.base/sun.net.www.protocol.https.HttpsClient.New(HttpsClient.java:372)
	at java.base/sun.net.www.protocol.https.AbstractDelegateHttpsURLConnection.getNewHttpClient(AbstractDelegateHttpsURLConnection.java:177)
	at java.base/sun.net.www.protocol.http.HttpURLConnection.plainConnect0(HttpURLConnection.java:1194)
	at java.base/sun.net.www.protocol.http.HttpURLConnection.plainConnect(HttpURLConnection.java:1082)
	at java.base/sun.net.www.protocol.https.AbstractDelegateHttpsURLConnection.connect(AbstractDelegateHttpsURLConnection.java:163)
	at java.base/sun.net.www.protocol.http.HttpURLConnection.getOutputStream0(HttpURLConnection.java:1375)
	at java.base/sun.net.www.protocol.http.HttpURLConnection.getOutputStream(HttpURLConnection.java:1350)
	at java.base/sun.net.www.protocol.https.HttpsURLConnectionImpl.getOutputStream(HttpsURLConnectionImpl.java:220)
	at org.sonatype.ossindex.service.client.transport.HttpUrlConnectionTransport.post(HttpUrlConnectionTransport.java:90)
	at org.sonatype.ossindex.service.client.internal.OssindexClientImpl.doRequestComponentReports(OssindexClientImpl.java:204)
	at org.sonatype.ossindex.service.client.internal.OssindexClientImpl.requestComponentReports(OssindexClientImpl.java:170)
	at org.owasp.dependencycheck.analyzer.OssIndexAnalyzer.requestReports(OssIndexAnalyze
[odc.log](https://github.com/jeremylong/DependencyCheck/files/7889528/odc.log)
r.java:212)
	at org.owasp.dependencycheck.analyzer.OssIndexAnalyzer.analyzeDependency(OssIndexAnalyzer.java:140)
	... 7 common frames omitted
2022-01-18 16:42:40,481 org.owasp.dependencycheck.utils.Settings:817
```


Attaching latest odc log
[odc.log](https://github.com/jeremylong/DependencyCheck/files/7889533/odc.log)

**Comment by aikebah** [comment]
@milyas-salik 
```
DEBUG - 
org.owasp.dependencycheck.analyzer.exception.AnalysisException: Failed to request component-reports
	at org.owasp.dependencycheck.analyzer.OssIndexAnalyzer.analyzeDependency(OssIndexAnalyzer.java:151)
	
...
Caused by: java.net.SocketTimeoutException: Connect timed out
	at java.base/sun.nio.ch.NioSocketImpl.timedFinishConnect(NioSocketImpl.java:546)
	at java.base/sun.nio.ch.NioSocketImpl.connect(NioSocketImpl.java:597)
	at java.base/java.net.Socket.connect(Socket.java:648)
	at java.base/sun.net.NetworkClient.doConnect(NetworkClient.java:177)
	at java.base/sun.net.www.http.HttpClient.openServer(HttpClient.java:474)
	at java.base/sun.net.www.http.HttpClient.openServer(HttpClient.java:569)
	at java.base/sun.net.www.protocol.https.HttpsClient.<init>(HttpsClient.java:265)
	at java.base/sun.net.www.protocol.https.HttpsClient.New(HttpsClient.java:372)
	at 
...
org.sonatype.ossindex.service.client.internal.OssindexClientImpl.doRequestComponentReports(OssindexClientImpl.java:204)
	at org.sonatype.ossindex.service.client.internal.OssindexClientImpl.requestComponentReports(OssindexClientImpl.java:170)
	at org.owasp.dependencycheck.analyzer.OssIndexAnalyzer.requestReports(OssIndexAnalyzer.java:212)
	at org.owasp.dependencycheck.analyzer.OssIndexAnalyzer.analyzeDependency(OssIndexAnalyzer.java:140)
	... 7 common frames omitted
```

shows that your issue still is with the network connectivity to the OSSINDEX

**Comment by milyas-salik** [comment]
Thanks @jeremylong , the accessibility issues were resolved by our network team and installing the .NET core run time resolved the assembly analyzer issue.

Thanks again to each one of you for the support !!

**Comment by milyas-salik** [comment]
Dear @jeremylong ,

Suddenly I started facing **Could not execute .NET AssemblyAnalyzer** issue, I checked .NET run time already installed and was working earlier with the results.

Attaching the odc.log, pls advise.
[Uploading odc.log…]()

See my .NET RUN times installed earlier.
![MicrosoftTeams-image (14)](https://user-images.githubusercontent.com/79843909/158177195-1367f594-c340-43a0-bdcc-e293a376c27a.png)

**It looks the issue started from 7.0.0 version until  6.5.3 it was perfect. Last time we installed .NET run time version 3.1.22.**
